tests/dashbio_demos/app_ideogram.py

```diff
@@ -14,833 +15,750 @@
-        elif value == "overlay-2":
-            answer = "https://eweitz.github.io/ideogram/data/annotations/1000_virtual_snvs.json"
-        return answer
+    def change_homology_ideogram(
+            chr_selected_1,
+            chr_selected_2,
+            start_one,
+            stop_one,
+            start_two,
+            stop_two,
+            start_one_a,
+            stop_one_a,
+            start_two_a,
+            stop_two_a,
+            current
+    ):
+        if current is None:
+            current = ideograms_initial['homology']
+        current.update(
+            chromosomes=[chr_selected_1, chr_selected_2],
+            homology={
+                'chrOne': {
+                    'organism': '9606',
+                    'start': [start_one, start_two],
+                    'stop': [stop_one, stop_two]
+                },
+                'chrTwo': {
+                    'organism': '9606',
+                    'start': [start_one_a, start_two_a],
+                    'stop': [stop_one_a, stop_two_a]
+                }
+            }
+        )
 
-    # Assembly
+        return current
 
     @app.callback(
-        Output("ideo-annotations",
-               "assembly"), [Input("annotation-select", "value")]
+        Output('brush-ideo-data', 'data'),
+        [Input('chr-brush', 'value')],
+        state=[State('brush-ideo-data', 'data')]
     )
-    def annot_assembly(value):
-        if value == "histogram":
-            return "GRCh37"
-        return None
+    def change_brush_ideogram(brush_value, current):
+        if current is None:
+            current = ideograms_initial['brush']
+        if brush_value is None:
+            brush_value = '1'
+        current.update(
+            chromosomes=[str(brush_value)],
+            brush='chr{}:1-10000000'.format(brush_value)
+        )
+        return current
 
     @app.callback(
-        Output("ideo-annotations", "annotationTracks"),
-        [Input("annotation-select", "value")],
+        Output('ideo-annotations-data', 'data'),
+        [Input('annotation-select', 'value'),
+         Input('bar-input', 'value'),
+         Input('orientation-anote', 'value'),
+         Input('color-input', 'value'),
+         Input('height-input', 'value')],
+        state=[State('ideo-annotations-data', 'data')]
     )
-    def annot_tracks(value):
-        if value == "overlay-2":
-            data = [
+    def change_annotation_ideogram(
+            annotation_select,
+            bar_input,
+            orientation_input,
+            color_input,
+            height_input,
+            current
+    ):
+        if current is None:
+            current = ideograms_initial['annotations']
+
+        annotations_layout = ''
+        annotations_path = None
+        annotations_assembly = None
+        annotation_tracks = None
+        annotation_height = height_input if height_input != '' else None
```

Comment:
Looks loose... I would expect something slightly stricter than just not being the empty string to be a proper `annotation_height` (otherwise you might as well just write `annotation_height = height_input if height_input else None`).

Reply:
Fixed in c02305e!

Reply:
Sure? No need to check for, say, the type of `height_input`?

Reply:
Nope, since it is always coming from a slider component, which always returns a number. Previously, the value was read from a text input component, which could be a string.